text/utilities/0000-health-checker-buildpack.md

```diff
@@ -0,0 +1,106 @@
+# Health Checker Support
+
+## Summary
+
+Introduce a Health Checker buildpack. This buildpack will provide access to one or more health checkers, where a health checker is a small binary that will be installed into the container image for the purpose of validating the health of the runnning image.
+
+By default, we'll provide a tiny HTTP health checker which can be used to probe a port and path. An HTTP 200 OK response returns exit code 0 and a non-2xx response returns 1.
+
+The Health Checker buildpack would be maintained by the existing [Utilities team](https://github.com/orgs/paketo-buildpacks/teams/utilities). It uses libpak & libcnb.
+
+## Motivation
```

Comment:
I am not sure that I am understanding a motive or use for this beyond something like integration testing an image with no app on it. I would love another use case for this if you have one at hand.

Reply:
Health check is a feature of your platform, like K8s or Cloud Foundry. Support from platform to platform varies. For example, K8s has HTTP, TCP and process health checks (it's getting/got gRPC health checks). What you get with other platforms will vary.

The primary focus for this is when running with Docker or Docker Compose. Docker only provides process health checks. Anything other health checks are done through a binary that must be run and return exit 0 (OK), 1 (FAIL). That is typically `curl` or `wget` for HTTP health checks, but those binaries are not in our Paketo base/tiny images nor do we want them to be, it adds a lot of surface area for CVEs.

The motivation here is to have a buildpack that provides custom-purpose health check binaries that can be used in Docker or on other platforms where health check support is not as extensive.